src/test/java/seedu/address/logic/parser/NuSchedulerParserTest.java

```diff
@@ -28,11 +24,12 @@ public class NuSchedulerParserTest {
     private final NuSchedulerParser parser = new NuSchedulerParser();
 
     @Test
-    public void parseCommand_addProfile() throws Exception {
+    public void parseCommand_profile() throws Exception {
         Profile profile = new ProfileBuilder().build();
+        String addProfileArguments = ProfileUtil.getAddProfileCommand(profile);
         AddProfileCommand command =
-                (AddProfileCommand) parser.parseCommand(ProfileUtil.getAddProfileCommand(profile));
-        assertEquals(new AddProfileCommand(profile), command);
+                (AddProfileCommand) parser.parseCommand(ProfileCommand.COMMAND_WORD + addProfileArguments);
+        assertEquals(new ProfileCommandParser().parse(addProfileArguments), command);
```

Comment:
Not sure if I should use ProfileCommandParser() here or just use AddProfileCommand() directly. I did it this way cause it sort of show the intermediate step.

Reply:
I feel like this test case should not be exposing the internals of `ProfileCommandParser`, hence `AddProfileCommand` shouldn't be used here. I think the test case should be something like
```java
assertTrue(parser.parseCommand(
                ProfileCommand.COMMAND_WORD + " -a n/A p/123 e/mail@example.com") instanceof ProfileCommand);
```
Admittedly, it's not a very elegant solution either.

Reply:
I think that might be possible, but I can't quite envision how to do so, because we would need the `NuScheduler` class to return this stub.

Reply:
I think what you suggested might be a good solution. It follows the other test cases in NuSchedulerParserTest. 👍